File: android/app/src/main/java/com/example/app/domain/navcaddy/models/Club.kt
package com.example.app.domain.navcaddy.models

import com.google.gson.annotations.SerializedName
import java.util.UUID

/**
 * Domain model representing a golf club with its specifications.
 *
 * Extended to support bag management with distance calibration and miss bias tracking.
 * All distances are in meters (metric units).
 *
 * Spec reference: player-profile-bag-management.md R2
 * Plan reference: player-profile-bag-management-plan.md Task 2
 *
 * @property id Unique identifier for this club
 * @property name Display name (e.g., "7-iron", "Driver", "PW")
 * @property type Club type category
 * @property loft Loft angle in degrees (optional)
 * @property distance Typical distance (optional) - DEPRECATED: use estimatedCarry
 * @property estimatedCarry User's estimated carry distance in meters
 * @property inferredCarry Carry distance inferred from shot data in meters (nullable until calculated)
 * @property inferredConfidence Confidence in the inferred value (0.0-1.0, nullable until calculated)
 * @property missBias Miss bias characteristics for this club (nullable until set)
 * @property shaft Shaft model/brand (optional)
 * @property flex Shaft flex (e.g., "Stiff", "Regular", "Senior")
 * @property notes User notes about this club
 */
data class Club(
    @SerializedName("id")
    val id: String = UUID.randomUUID().toString(),

    @SerializedName("name")
    val name: String,

    @SerializedName("type")
    val type: ClubType,

    @SerializedName("loft")
    val loft: Double? = null,

    @SerializedName("distance")
    val distance: Int? = null,

    // New fields for bag management (player-profile-bag-management.md R2, R3, R4)
    @SerializedName("estimated_carry")
    val estimatedCarry: Int,

    @SerializedName("inferred_carry")
    val inferredCarry: Int? = null,

    @SerializedName("inferred_confidence")
    val inferredConfidence: Double? = null,

    @SerializedName("miss_bias")
    val missBias: MissBias? = null,

    @SerializedName("shaft")
    val shaft: String? = null,

    @SerializedName("flex")
    val flex: String? = null,

    @SerializedName("notes")
    val notes: String? = null
) {
    init {
        require(estimatedCarry > 0) { "Estimated carry must be positive" }
        inferredConfidence?.let {
            require(it in 0.0..1.0) { "Inferred confidence must be between 0 and 1" }
        }
    }
}

File: ios/App/Sources/Core/Domain/NavCaddy/Club.swift
import Foundation

/// Represents a golf club with its characteristics.
///
/// All distances are in meters (metric units).
///
/// Spec reference: player-profile-bag-management.md R2
/// Plan reference: player-profile-bag-management-plan.md Task 2
struct Club: Codable, Hashable, Identifiable {
    let id: String
    let name: String
    let type: ClubType
    let loft: Double?
    let distance: Int?

    // MARK: - Bag Management Fields (Task 2)

    /// User's estimated carry distance in meters.
    let estimatedCarry: Int
    /// Carry distance inferred from shot data in meters (nil until calculated).
    let inferredCarry: Int?
    /// Confidence level for inferred carry (0.0-1.0).
    let inferredConfidence: Double?
    /// Miss bias tendency for this club.
    let missBias: MissBias?
    /// Shaft specification (e.g., "Graphite", "Steel").
    let shaft: String?
    /// Flex specification (e.g., "Stiff", "Regular").
    let flex: String?
    /// User notes about the club.
    let notes: String?

    init(
        id: String = UUID().uuidString,
        name: String,
        type: ClubType,
        loft: Double? = nil,
        distance: Int? = nil,
        estimatedCarry: Int,
        inferredCarry: Int? = nil,
        inferredConfidence: Double? = nil,
        missBias: MissBias? = nil,
        shaft: String? = nil,
        flex: String? = nil,
        notes: String? = nil
    ) {
        precondition(estimatedCarry > 0, "Estimated carry must be positive")
        if let conf = inferredConfidence {
            precondition(
                (0.0...1.0).contains(conf),
                "Inferred confidence must be between 0 and 1"
            )
        }
        self.id = id
        self.name = name
        self.type = type
        self.loft = loft
        self.distance = distance
        self.estimatedCarry = estimatedCarry
        self.inferredCarry = inferredCarry
        self.inferredConfidence = inferredConfidence
        self.missBias = missBias
        self.shaft = shaft
        self.flex = flex
        self.notes = notes
    }
}
